cupy/core/reduction.pxi

```diff
@@ -173,36 +173,17 @@ cpdef (Py_ssize_t, Py_ssize_t, Py_ssize_t) _get_block_specs(  # NOQA
 cdef Py_ssize_t _block_size = 256 if runtime._is_hip_environment else 512
 
 
-cdef tuple _get_reduction_args(
-        list in_args, list out_args, tuple in_params, tuple out_params,
-        tuple axis_permutes, tuple a_shape, tuple out_shape,
-        bint reduce_dims):
-    # Returns a tuple that contains following items
-    # - list of arguments passed to the __global__ function.
-    # - block_size
-    # - out_block_num
-    cdef Py_ssize_t contiguous_size, block_size, block_stride, out_block_num
-    in_shape = _set_permuted_args(
-        in_args, axis_permutes, a_shape, in_params)
-    contiguous_size = _get_contiguous_size(
-        in_args, in_params, len(in_shape), len(out_shape))
-
-    if reduce_dims:
-        in_shape = _reduce_dims(in_args, in_params, in_shape)
-        out_shape = _reduce_dims(out_args, out_params, out_shape)
-
-    block_size, block_stride, out_block_num = _get_block_specs(
-        internal.prod_sequence(in_shape),
-        internal.prod_sequence(out_shape),
-        contiguous_size)
-
+cdef list _get_reduction_args(
+        list in_args, list out_args,
+        tuple in_shape, tuple out_shape,
+        Py_ssize_t block_stride):
+    # Returns a list of arguments passed to the __global__ function.
     in_indexer = Indexer(in_shape)
     out_indexer = Indexer(out_shape)
-
-    # The last argument is always block_stride.
     s = _scalar.CScalar_from_int32(block_stride)
-    return (in_args + out_args + [in_indexer, out_indexer, s],
-            block_size, out_block_num)
+    return (
```

Comment:
Just a nitpick but how about removing this redundant parenthesis?

Reply:
Instead I inlined this function in bf71432abde1e1ea62bcbdafefa0ff9b02e2d897.
Please take a look.